lib/posix/pthread.c

```diff
@@ -205,13 +212,13 @@ int pthread_attr_setstack(pthread_attr_t *_attr, void *stackaddr, size_t stacksi
 
 static bool pthread_attr_is_valid(const struct pthread_attr *attr)
 {
-	/*
-	 * FIXME: Pthread attribute must be non-null and it provides stack
-	 * pointer and stack size. So even though POSIX 1003.1 spec accepts
-	 * attrib as NULL but zephyr needs it initialized with valid stack.
-	 */
-	if (attr == NULL || attr->initialized == 0U || attr->stack == NULL ||
-	    attr->stacksize == 0) {
+	/* auto-alloc thread stack */
+	if (attr == NULL) {
+		return true;
+	}
+
+	/* caller-provided thread stack */
+	if (attr->initialized == 0U || attr->stack == NULL || attr->stacksize == 0) {
```

Comment:
Looks like this requires that any use of thread attributes must include a pre-allocated stack? Any particular reason for this restriction? Seems like applications might need to use thread attributes for some other reason?

Reply:
`attr == NULL` is valid (just a few lines above here)

Reply:
right, but if the application provides attributes, those must include a stack. Seems like it might be nice to allow applications to provide attributes for some other reason and still have an automatically allocated stack.

Reply:
Maybe - I would say, if there is a corner case we should have some testing and code review though.